bears/general/SpaceConsistencyBear.py

```diff
@@ -38,7 +41,52 @@ def run(self,
         result_texts = []
         additional_info_texts = []
 
-        for line_number, line in enumerate(file, start=1):
+        def get_blanklines_nr_start():
+            line_nr_start = False
+            if not allow_trailing_blanklines:
```

Comment:
I think this should be `if allow_trailing_blanklines:` because only then you will have to check `if replacement.strip() == ''` otherwise `line_nr_start` [which I am guessing to be the *start from where blank lines or in case of trailing spaces allowed blank lines (with spaces) start]

Reply:
I stand a little bit corrected, if you `if` condition is true `line_nr_start` would be just the last line or False (in case there is no new line at EOF), if your `if` condition is incorrect then your `line_nr_start` would again be just the last line or False [in case there is no new line at EOF (even with spaces)] so don't quite get what you are trying to do over here

Reply:
note there are 5 cases in which the file can end in case of trailing spaces are allowed